packages/design-system/src/components/tree/tree.tsx

```diff
@@ -44,6 +44,8 @@ export type TreeProps<Data extends { id: string }> = {
     dropTarget: { itemSelector: ItemSelector; position: number | "end" };
   }) => void;
   onCancel: () => void;
+  isEditingItemName: ItemId | undefined;
+  onEditItemName: (itemId: ItemId) => void;
```

Comment:
semantics of this function are not clear, this is from what I can see starting the editing mode:
`onItemEditingStart`